### PR TITLE
[Kernel/Memory] Do not clear range if it was previously commited

### PR DESCRIPTION
**Pull request by Gliniak**
Related to issue: #1667 

Most of the info is included in that issue (Including example code). I couldn't add such check when we're not providing ``*base_addr_ptr`` from obvious reason. However I wouldn't expect that to be such a big deal.

Note for testers please check if this PR causes any regression based on xenia: https://github.com/xenia-project/xenia/commit/ff5c5f01e0e31abee675be7e48c11c20642a40fb

Affected Titles:
- Serious Sam: TFE
- Perfect Dark Zero

### CODE REVIEW

File: src/xenia/kernel/xboxkrnl/xboxkrnl_memory.cc [hunk 1]
```diff
@@ -136,13 +136,15 @@ dword_result_t NtAllocateVirtualMemory(lpdword_t base_addr_ptr,
   uint32_t protect = FromXdkProtectFlags(protect_bits);
   uint32_t address = 0;
   BaseHeap* heap;
+  HeapAllocationInfo* prev_alloc_info = new HeapAllocationInfo();
```
Comment (gibbed) [inline src/xenia/kernel/xboxkrnl/xboxkrnl_memory.cc:139]: You are leaking memory.

`HeapAllocationInfo prev_alloc_info = {};`

`heap->QueryRegionInfo(adjusted_base, &prev_alloc_info);`

etc

**Review by gibbed — CHANGES_REQUESTED** [review]

File: src/xenia/kernel/xboxkrnl/xboxkrnl_memory.cc (hunk 1 above)
Comment (gibbed) [reply src/xenia/kernel/xboxkrnl/xboxkrnl_memory.cc:139]: I'd say it's probably best as something like:

```c
HeapAllocationInfo prev_alloc_info = {};
bool was_commit = heap->QueryRegionInfo(adjusted_base, &prev_alloc_info) && (prev_alloc_info.kMemoryAllocationCommit & kMemoryAllocationCommit) != 0;
```

Then later use `if (!was_commit)`.

Comment (Gliniak) [reply src/xenia/kernel/xboxkrnl/xboxkrnl_memory.cc:139]: I disagree with second post. 
I would need to make ``bool was_commit`` set it from default to false and then inside branches aka:
 ```if (adjusted_base != 0) {```
get that info and store it to that boolean, but at this point we have it already in prev_alloc_info, so imo usage of ``bool was_commit`` is quite unnecessary.

I will just add note that should resolve eventual confusion

Comment (gibbed) [reply src/xenia/kernel/xboxkrnl/xboxkrnl_memory.cc]: The point here is to never used an uninitialized `prev_alloc_info`; even if we initialize it to zero, there's no true guarantee as to what it is in `prev_alloc_info` if `QueryRegionInfo` returns `false`.

Comment (Gliniak) [reply src/xenia/kernel/xboxkrnl/xboxkrnl_memory.cc]: Resolved

**Pull request merged**